src/components/reusable/timepicker/timepicker.ts

```diff
@@ -0,0 +1,251 @@
+  /** Optional text beneath the input. */
+  @property({ type: String })
+  caption = '';
+
+  /** The value of the time input is always in 24-hour format that includes leading zeros: hh:mm,
+   *  regardless of the input format, which is likely to be selected based on the user's locale (or by the user agent).
+   *  If the time includes seconds (by step attribute), the format is always hh:mm:ss */
+  @property({ type: String })
+  value = '';
+
+  /** Time input name. */
+  @property({ type: String })
+  name = '';
+
+  /** Makes the input required. */
+  @property({ type: Boolean })
+  required = false;
+
+  /** Input disabled state. */
+  @property({ type: Boolean })
+  disabled = false;
+
+  /** Time input invalid text. */
+  @property({ type: String })
+  invalidText = '';
+
+  /** Time input warn text. */
+  @property({ type: String })
+  warnText = '';
+
+  /** Maximum time in hh:mm or hh:mm:ss format only must always greater than min. */
+  @property({ type: String })
+  maxTime = '';
+
+  /** Minimum time. in hh:mm or hh:mm:ss format only & must always lesser than max*/
+  @property({ type: String })
+  minTime = '';
+
+  /** Specifies the granularity that the value must adhere to, or the special value any,
+   * It takes value that equates to the number of seconds you want to increment by;
+   * the default (60 sec.). If you specify a value of less than 60 sec., the time input will show a seconds input area alongside the hours and minutes */
+  @property({ type: String })
+  step = '';
+
+  /**
+   * Queries the <input> DOM element.
+   * @ignore
+   */
+  @query('input')
+  inputEl!: HTMLInputElement;
+
+  regxPatternWithSec = /^\d{2}:\d{2}:\d{2}$/; // hh:mm:ss
+  regxPatternWithoutSec = /^\d{2}:\d{2}$/; // hh:mm
+
+  override render() {
+    return html`
+      <div class="time-picker" ?disabled=${this.disabled}>
+        <label class="label-text" for=${this.name}>
+          ${this.required ? html`<span class="required">*</span>` : null}
+          <slot></slot>
+        </label>
+
+        <div
+          class="${classMap({
+            'input-wrapper': true,
+          })}"
+        >
+          <input
+            class="${classMap({
+              'size--sm': this.size === 'sm',
+              'size--lg': this.size === 'lg',
+            })}"
+            type="time"
+            id=${this.name}
+            name=${this.name}
+            value=${this.value}
+            step=${ifDefined(this.step)}
+            ?required=${this.required}
+            ?disabled=${this.disabled}
+            ?invalid=${this.invalidText !== ''}
+            min=${ifDefined(this.minTime)}
+            max=${ifDefined(this.maxTime)}
+            @input=${(e: any) => this.handleInput(e)}
+          />
+
+          ${this.value !== ''
+            ? html`
+                <button class="clear" @click=${() => this.handleClear()}>
+                  <kd-icon .icon=${clearIcon}></kd-icon>
+                </button>
+              `
+            : null}
```

Comment:
I don't think the design for this component has a clear button, but I'll let you decide if you want to keep it.

Reply:
OK, just make sure there's right padding to account for that then.

Reply:
Still needs the right padding to make sure there's always space for the clear icon so it doesn't overlap the input content. I would suggest actually removing the min-width entirely so consumers can have more flexibility.

Reply:
with min-width it looks proper
<img width="265" alt="Screenshot 2023-08-30 at 9 22 11 PM" src="https://github.com/kyndryl-design-system/web-components/assets/140634142/fe0dc025-ede6-4baa-87f6-287db43da693">

Reply:
Right, this is why you need right padding to cover the area the clear button takes up.

Reply:
in UX there is left and right padding 16px. What you suggest? Should I decrease right padding. 
<img width="262" alt="Screenshot 2023-08-30 at 9 23 48 PM" src="https://github.com/kyndryl-design-system/web-components/assets/140634142/dfb3933b-060b-4a54-a8c1-965dc3381a6a">

Reply:
Copy text input which also has padding for the clear button.

Reply:
right now it's looking like below.
<img width="267" alt="Screenshot 2023-08-30 at 9 26 01 PM" src="https://github.com/kyndryl-design-system/web-components/assets/140634142/77720f99-19ff-4370-8968-72a44d304ec0">

Reply:
The padding-right value from text input is 48px

Reply:
yes. but for timepicker (in UX) its 16px left and right and if I gonna make padding-right: 48px for timepicker then native clock icon will also taking that padding. For ex: below image.
<img width="342" alt="Screenshot 2023-08-30 at 10 32 45 PM" src="https://github.com/kyndryl-design-system/web-components/assets/140634142/1f21ed88-3ffb-4931-ad17-aea2aa12bfa6">

Reply:
as for text input has type="text" and when we do type="time", Natively clock icon appears with input field. hence we need to keep equal padding at left and right side

Reply:
Done. please apporve.